jdi-light-mobile/src/main/java/com/epam/jdi/light/mobile/MobileUtils.java

```diff
@@ -0,0 +1,112 @@
+import static com.epam.jdi.tools.ReflectionUtils.*;
+import static com.epam.jdi.tools.StringUtils.namesEqual;
+
+/**
+ * This class introduces the wrappers around the driver methods.
+ * Since each mobile driver has its own set of features, every time some
+ * driver method is called, we need to perform a check if the current driver is able to perform
+ * the desired action. These methods perform such check and fire an exception if the method
+ * cannot be called with the current driver
+ */
+public class MobileUtils {
+    public static <I> void executeDriverMethod(Class<I> clazz, Consumer<I> consumer) {
+        WebDriver driver = getDriver();
+        if (clazz.isInstance(driver)) {
+            consumer.accept(clazz.cast(driver));
+        } else {
+            throw exception("Cannot use this method. The driver needs to extend/implement the " + clazz.getName());
+        }
+    }
+    public static <I, R> R executeDriverMethod(Class<I> clazz, Function<I, R> function) {
+        WebDriver driver = getDriver();
+        if (clazz.isInstance(driver)) {
+            return function.apply(clazz.cast(driver));
+        } else {
+            throw exception("Cannot use this method. The driver needs to extend/implement the " + clazz.getName());
+        }
+    }
+
+
+    public static JFunc2<Object, String, HasTouchActions> GET_DEFAULT_BUTTON =
+            (obj, buttonName) -> $(By.xpath(String.format("//XCUIElementTypeButton[@name='%s']", buttonName)),
+                    obj).setName(buttonName);
+
+    public static JFunc2<Object, String, HasTouchActions> GET_BUTTON = (obj, buttonName) -> {
+        List<Field> fields = getFields(obj, IsButton.class);
+        if (fields.size() == 0)
+            fields = getFieldsExact(obj, MobileAppUIElement.class, UIElement.class);
+        if (fields.size() > 1) {
+            fields = filter(fields, f ->
+                    isInterfaceAnd(getValueField(f, obj).getClass(), HasTouchActions.class, INamed.class));
+            if (fields.size() >= 1) {
+                Collection<HasTouchActions> buttons = select(fields,
+                        f -> (HasTouchActions) getValueField(f, obj));
+                HasTouchActions button = first(buttons, b -> namesEqual(toButton(((INamed) b).getName()), toButton(buttonName)));
+                if (button != null)
+                    return button;
+            }
+        }
+        if (fields.size() == 1) {
+            Field field = fields.get(0);
+            Object btnObj = getValueField(field, obj);
+            if (isInterface(btnObj.getClass(), HasTouchActions.class))
+                return (HasTouchActions) btnObj;
+        }
+        return GET_DEFAULT_BUTTON.execute(obj, buttonName);
+    };
+
+    private static String toButton(String buttonName) {
+        return buttonName.toLowerCase().contains("button") ? buttonName : buttonName + "button";
+    }
+
+    public static JFunc2<Object, String, HasTouchActions> GET_DEFAULT_APP_MENU_ITEM =
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6361799)